pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +755,97 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	copyNameNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "name"},
+	)
+	copyDiffNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "diff"},
+	)
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileName,
+		OnPress: func() error {
+			name := self.context().GetSelectedItemId()
+			if name == "" {
+				return self.c.ErrorMsg(copyNameNoItemSelectionErr)
+			}
+
+			return self.copyToClipboard(name)
+		},
+		Key: 'n',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label: self.c.Tr.CopySelectedDiff,
+		OnPress: func() error {
+			path := self.context().GetSelectedPath()
+			if path == "" {
+				return self.c.ErrorMsg(copyDiffNoItemSelectionErr)
+			}
+			diff, err := self.c.Git().Diff.GetPathDiff(path, self.hasPathStagedChanges())
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 's',
+	}
+
+	if self.context().GetSelected() == nil {
+		copyNameItem.DisabledReason = copyNameNoItemSelectionErr
+		copyFileDiffItem.DisabledReason = copyDiffNoItemSelectionErr
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.CopyToClipboardMenu,
+		Items: []*types.MenuItem{
+			copyNameItem,
+			copyFileDiffItem,
+			{
+				Label: self.c.Tr.CopyAllFilesDiff,
+				OnPress: func() error {
+					diff, err := self.c.Git().Diff.GetAllDiff(self.hasStagedChanges())
+					if err != nil {
+						return self.c.Error(err)
+					}
+					return self.copyToClipboard(diff)
+				},
+				Key: 'a',
+			},
+		},
+	})
+}
+
+func (self *FilesController) hasStagedChanges() bool {
+	files := self.context().GetAllFiles()
+	for _, file := range files {
+		if file.HasStagedChanges {
+			return true
+		}
+	}
+	return false
+}
+
+func (self *FilesController) hasPathStagedChanges() bool {
```

Comment:
@mark2185, I tried substituting this function for `self.getSelectedFile().HasStagedChanges`, but this only works for files. If you select a directory, `getSelectedFile()` returns `nil`. Are you happy to stick with this helper method?